### PR TITLE
[docs] Add models.md

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR adds documentation for `createModel()`.

### CODE REVIEW

File: docs/guides/models.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+In XState, you can model a machine's `context` and `events` externally by using `createModel(...)`. This provides a convenient way to strongly type `context` and `events`, as well as helpers for event creation, assignment and other implementation details in the future.
+
+## `createModel(...)`
+
+The `createModel(...)` function takes
+
+| Argument              | Type   | Description                                                |
+| --------------------- | ------ | ---------------------------------------------------------- |
+| `initialContext`      | object | The initial `context` value                                |
+| `creators` (optional) | object | An object containing various creators (currently `events`) |
```
Comment (dhirjgupta) [inline docs/guides/models.md:12]:
```suggestion
| `creators` (optional) | object | An object containing various creators (`events`) |
```
Makes it clearer without "currently" imo. Or perhaps (currently: `events`. coming soon:  [some known things on the road map])

File: docs/guides/models.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    // Event creators
+    events: {
+      updateName: (value: string) => ({ value }),
+      updateAge: (value: number) => ({ value }),
+      anotherEvent: () => ({}) // no payload
+    }
+  }
+);
+
+const machine = createMachine<typeof userModel>({
```
[the remark this answers is not in the document]
Comment (dhirjgupta) [reply docs/guides/models.md:71]: maybe something like "...a convenient utility that encourages a creator pattern and provides typescript benefits"

**Comment by fraserdrops** [comment]
I'm a little confused about what createModel _is_. I think the docs do a great job explaining the benefits of it, but conceptually I already thought I was creating a model when I was writing a machine - this feels like a model for a model, and I'm unsure why I need to do that (apart from because it makes things easier, which admittedly isn't a bad reason).

**Comment by erikras** [comment]
> - `createModel` creates a model for context and _events_ (and other things)

So it creates a model for the machine. Why not `createMachineModel`? Too repetitive?

**Comment by davidkpiano** [comment]
> > * `createModel` creates a model for context and _events_ (and other things)
> 
> So it creates a model for the machine. Why not `createMachineModel`? Too repetitive?

Too wordy, and also, it's useful outside of machines 😉
